### PR TITLE
make format_int constexpr

### PR DESCRIPTION
**Pull request by dixlorenz**

### CODE REVIEW

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR. Overall looks good but please apply clang-format.

File: include/fmt/format.h [hunk 1]
```diff
@@ -3933,32 +3935,38 @@ class format_int {
...
   /// character appended.
-  auto c_str() const -> const char* {
+  FMT_CONSTEXPR20 auto c_str() const -> const char* {
     buffer_[buffer_size - 1] = '\0';
     return str_;
   }
 
   /// Returns the content of the output buffer as an `std::string`.
-  auto str() const -> std::string { return std::string(str_, size()); }
+  FMT_CONSTEXPR20 auto str() const -> std::string {
```
Comment (vitaut) [inline include/fmt/format.h:3967]: Let's drop FMT_CONSTEXPR20 here since constexpr std::string is less widely available.

Comment (dixlorenz) [reply include/fmt/format.h]: ok, one can always write a constexpr free function for it (I did anyway)

**Review by vitaut — APPROVED** [review]

**Comment by vitaut** [comment]
Merged, thanks!

**Pull request merged**